### DISCUSSION TITLE
Custom JsonIgnore condition

### DISCUSSION THREAD

**Discussion by whoaskedfrfr** [dotnet/runtime#91793] [Q&A]
Hey,

Is there a way to customize JsonIgnore condition?

As in, fully developer controlled thing? My idea was some kind of interface with single method bool Ignore() that is overridable. Is there something like this?

**Comment by Pentadome** [comment]
There is no build-in way to do this afaik but you could write something like this yourself. Take a look at [Type info modifiers](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/custom-contracts#example-increment-a-propertys-value). For conditionally ignoring a property, you can set the [JsonPropertyInfo.ShouldSerialize ](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.metadata.jsonpropertyinfo.shouldserialize?view=net-7.0#system-text-json-serialization-metadata-jsonpropertyinfo-shouldserialize) property with your own custom logic.
